addons/web/static/tests/views/fields/datetime_field.test.js

```diff
@@ -497,3 +497,57 @@ test("datetime field: use picker with arabic numbering system", async () => {
     await animationFrame();
     expect(queryFirst("[name=datetime] input")).toHaveValue("٠٢/٠٨/٢٠١٧ ١١:٤٥:٠٠");
 });
+
+test("datetime field in list view with showSeconds option", async () => {
+    mockTimeZone(+2);
+    onRpc("has_group", () => true);
+
+    await mountView({
+        type: "list",
+        resModel: "partner",
+        arch: /* xml */ `
+            <tree>
+                <field name="datetime" widget="datetime" options="{'showSeconds': false}" string="showSeconds as false"/>
+                <field name="datetime" widget="datetime" options="{'showSeconds': true}" string="showSeconds as true"/>
+            </tree>`,
+    });
+
+    const [dateField1, dateField2] = queryAll(".d-flex.gap-2.align-items-center");
+    const expectedDateString1 = "02/08/2017 12:00";
+    const expectedDateString2 = "02/08/2017 12:00:00";
+
+    expect(dateField1).toHaveText(expectedDateString1, {
+        message: "seconds should be hidden for showSeconds false",
+    });
+
+    expect(dateField2).toHaveText(expectedDateString2, {
+        message: "seconds should be visible for showSeconds true",
+    });
+});
+
+test("datetime field in form view with showSeconds option", async () => {
+    mockTimeZone(+2);
+
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        arch: /* xml */ `
+            <form>
+                <field name="datetime" widget="datetime" options="{'showSeconds': false}" string="showSeconds as false"/>
+                <field name="datetime" widget="datetime" options="{'showSeconds': true}" string="showSeconds as true"/>
+            </form>`,
+    });
+
+    const [dateField1, dateField2] = queryAll(".o_input.cursor-pointer");
+    click(dateField1)
+    edit("02/08/2017 11:00:00", { confirm: "Enter" });
+    await animationFrame();
```

Comment:
you should also assert that seconds aren't displayed in the picker

Reply:
@Polymorphe57 I wanted to clarify one thing: in the task, I haven't made any changes to the picker itself. The seconds are only shown in the picker when the rounding is set to 0. If rounding is applied, the seconds are not shown by default, meaning seconds will be visible in the picker even if the `show_seconds option` is set to `false` but the rounding is set to `0`.

Should I assert this in the test case, as it is not related to my improvement? Or should I make the improvement to forever hide the seconds from the picker when the `showSeconds `is set to `false`.

Reply:
when show_seconds=false, I think you should set rounding to 0 if not set explicitly to another value. This way you allow the user to see seconds (if any) and modify them. So maybe we should assert that we **have** seconds in the picker here. The point is that we do not want the user to be unable to modify seconds.
CTH agrees with me on that